### PR TITLE
chore: properly evaluate how tests are skipped on CI when checking modified files

### LINKED ISSUE #1915: Unit test stage is always skipped with "Waiting for run to start"
CI Build: https://beats-ci.elastic.co/blue/organizations/jenkins/e2e-tests%2Fe2e-testing-mbp/detail/PR-1914/3/pipeline/118

### PR DESCRIPTION
**Pull request by mdelapenya**
## What does this PR do?
This PR does two things:

1. wraps the evaluation of the stack deployment stage in an `anyOf` block, so that is consistent with all other stages.
2. extracts the logic to evaluate if the tests must be skipped to a single function. This function starts setting the `SKIP_TESTS` env var to false, and:
  - if there are only markdown files in the changeset, the it will set the env var `SKIP_TESTS` to true.
  - if the changeset contains at least one modified file matching certain paths (see the modified files in this PR), then it will set the `SKIP_TESTS` env var to false.

## Why is it important?
The unit tests stage is constantly skipped.

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have run the Unit tests (`make unit-test`), and they are passing locally
- [ ] I have run the End-2-End tests for the suite I'm working on, and they are passing locally
- [ ] I have noticed new Go dependencies (run `make notice` in the proper directory)

## Related issues

- Fixes #1915

### CODE REVIEW

File: .ci/Jenkinsfile [hunk 1]
```diff
@@ -93,13 +93,8 @@ pipeline {
...
             setEnvVar("GO_VERSION", readFile("${env.WORKSPACE}/${env.BASE_DIR}/.go-version").trim())
-            dir("${BASE_DIR}"){
-              // Skip all the test stages for PR's with markdown changes only
-              setEnvVar("SKIP_TESTS", isGitRegionMatch(patterns: [ '.*\\.md' ], shouldMatchAll: true))
-              script {
-                def regexps = [ "^e2e/_suites/fleet/.*", "^e2e/_suites/helm/.*", "^e2e/_suites/kubernetes-autodiscover/.*", "^.ci/.*", "^cli/.*", "^e2e/.*\\.go", "^internal/.*\\.go" ]
-                setEnvVar("SKIP_TESTS", !isGitRegionMatch(patterns: regexps, shouldMatchAll: false))
-              }
+            script {
+              checkSkipTest()
```
Comment (mdelapenya) [inline .ci/Jenkinsfile:97]:
```suggestion
              checkSkipTests()
```

File: .ci/Jenkinsfile [hunk 2]
```diff
@@ -305,6 +302,25 @@ pipeline {
   }
 }
 
+// this function evaluates if the test stage of the build must be executed
+def checkSkipTest() {
```
Comment (mdelapenya) [inline .ci/Jenkinsfile:306]:
```suggestion
def checkSkipTests() {
```

File: .ci/Jenkinsfile [hunk 3]
```diff
@@ -93,13 +93,8 @@ pipeline {
...
             stash allowEmpty: true, name: 'source', useDefaultExcludes: false
             setEnvVar("GO_VERSION", readFile("${env.WORKSPACE}/${env.BASE_DIR}/.go-version").trim())
-            dir("${BASE_DIR}"){
-              // Skip all the test stages for PR's with markdown changes only
-              setEnvVar("SKIP_TESTS", isGitRegionMatch(patterns: [ '.*\\.md' ], shouldMatchAll: true))
-              script {
-                def regexps = [ "^e2e/_suites/fleet/.*", "^e2e/_suites/helm/.*", "^e2e/_suites/kubernetes-autodiscover/.*", "^.ci/.*", "^cli/.*", "^e2e/.*\\.go", "^internal/.*\\.go" ]
-                setEnvVar("SKIP_TESTS", !isGitRegionMatch(patterns: regexps, shouldMatchAll: false))
-              }
+            script {
```
Comment (v1v) [inline .ci/Jenkinsfile:96]: NIT:
IIRC, this unrequired when running a method

File: .ci/Jenkinsfile [hunk 4]
```diff
@@ -305,6 +302,25 @@ pipeline {
   }
 }
 
+// this function evaluates if the test stage of the build must be executed
+def checkSkipTests() {
+  dir("${BASE_DIR}"){
+    // running the test should be mandatory by default
+    setEnvVar("SKIP_TESTS", false)
```
Comment (v1v) [inline .ci/Jenkinsfile:309]: Unrequired, since L319 already provides the value?

```suggestion
```

**Review by v1v — APPROVED** [review]
My comments are NIT

File: .ci/Jenkinsfile (hunk 3 above)
Comment (kuisathaverat) [reply .ci/Jenkinsfile:96]: yep, not needed

**Review by kuisathaverat — APPROVED** [review]

File: .ci/Jenkinsfile [hunk 5]
```diff
@@ -93,13 +93,8 @@ pipeline {
...
-            dir("${BASE_DIR}"){
-              // Skip all the test stages for PR's with markdown changes only
-              setEnvVar("SKIP_TESTS", isGitRegionMatch(patterns: [ '.*\\.md' ], shouldMatchAll: true))
-              script {
-                def regexps = [ "^e2e/_suites/fleet/.*", "^e2e/_suites/helm/.*", "^e2e/_suites/kubernetes-autodiscover/.*", "^.ci/.*", "^cli/.*", "^e2e/.*\\.go", "^internal/.*\\.go" ]
-                setEnvVar("SKIP_TESTS", !isGitRegionMatch(patterns: regexps, shouldMatchAll: false))
-              }
+            script {
+              checkSkipTests()
             }
```
Comment (mdelapenya) [inline .ci/Jenkinsfile:98]:
```suggestion
            checkSkipTests()
```

**Pull request merged**